launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -528,6 +528,7 @@ public SparkAppHandle startApplication(SparkAppHandle.Listener... listeners) thr
     pb.environment().put(LauncherProtocol.ENV_LAUNCHER_PORT,
       String.valueOf(LauncherServer.getServerInstance().getPort()));
     pb.environment().put(LauncherProtocol.ENV_LAUNCHER_SECRET, handle.getSecret());
+    pb.environment().put(LauncherProtocol.ENV_LAUNCHER_STOP_FLAG, String.valueOf(stopIfInterrupted));
```

Comment:
it would be nice if we just had one mechanism env or configs, could you look at making these configs?